expr/parse.go

```diff
@@ -133,25 +148,30 @@ func ParseMany(targets []string) ([]*expr, error) {
 }
 
 func skipWhitespace(s string) string {
-	for len(s) > 1 && s[0] == ' ' {
+	for len(s) >= 1 && s[0] == ' ' {
 		s = s[1:]
 	}
 	return s
 }
 
 // Parses an expression string and turns it into an expression
 // also returns any leftover data that could not be parsed
-// piped means: the expression to be parsed is known to receive a piped input
-// (should only be set by internal calls)
-func Parse(e string, piped bool) (*expr, string, error) {
+func Parse(e string, pCtx ParseContext) (*expr, string, error) {
 	e = skipWhitespace(e)
 
 	if len(e) == 0 {
 		return nil, "", ErrMissingExpr
 	}
 
-	if '0' <= e[0] && e[0] <= '9' || e[0] == '-' || e[0] == '+' {
-		return parseConst(e)
+	if pCtx.IsFullArg && ('0' <= e[0] && e[0] <= '9' || e[0] == '-' || e[0] == '+') {
+		constExpr, leftover, err := parseConst(e)
```

Comment:
unrelated to your PR, but this function is kinda pooorly named. maybe we should name this `parseNumber`.
and akin to `isFnStartChar` maybe we could have `isNumberStartChar`

Reply:
Renamed `parseConst` and extracted the start char check to `isNumStartChar` in https://github.com/grafana/metrictank/pull/2012/commits/f8f8b1ba90b6849aeb01b9920126ee9fee28222b